macOS: support version 14.0 and later

We are currently supporting [macOS 10.15](https://en.wikipedia.org/wiki/MacOS_version_history#:~:text=July%2021%2C%202021-,macOS%2010.15,-Catalina) and later (which was released in 2019 and is not supported anymore). We should therefore aim at supporting the oldest supported version of macOS, which is currently [macOS 14](https://en.wikipedia.org/wiki/MacOS_version_history#:~:text=August%2020%2C%202025-,macOS%2014,-Sonoma).

Right now, if we were to try to build libCellML on an up-to-date version of macOS (i.e. 26.5.1) then we would get things like:
```
FAILED: [code=1] src/CMakeFiles/cellml.dir/entity.cpp.o
/usr/local/bin/buildcache /opt/homebrew/opt/llvm/bin/clang++ -Dcellml_EXPORTS -I/Users/Alan/libCellML/build/src/api -I/Users/Alan/libCellML/src/api -I/Users/Alan/libCellML/src/api/libcellml/module -I/Users/Alan/libCellML/build/src -isystem /Library/Developer/CommandLineTools/SDKs/MacOSX26.sdk/usr/include/libxml2 -g -std=gnu++20 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.15 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -fprofile-arcs -ftest-coverage -fprofile-instr-generate -fcoverage-mapping -Wall -W -Werror -MD -MT src/CMakeFiles/cellml.dir/entity.cpp.o -MF src/CMakeFiles/cellml.dir/entity.cpp.o.d -o src/CMakeFiles/cellml.dir/entity.cpp.o -c /Users/Alan/libCellML/src/entity.cpp
In file included from /Users/Alan/libCellML/src/entity.cpp:17:
In file included from /Users/Alan/libCellML/src/api/libcellml/entity.h:19:
In file included from /opt/homebrew/Cellar/llvm/22.1.8/bin/../include/c++/v1/string:594:
In file included from /opt/homebrew/Cellar/llvm/22.1.8/bin/../include/c++/v1/__algorithm/max.h:12:
In file included from /opt/homebrew/Cellar/llvm/22.1.8/bin/../include/c++/v1/__algorithm/comp.h:12:
In file included from /opt/homebrew/Cellar/llvm/22.1.8/bin/../include/c++/v1/__config:15:
/opt/homebrew/Cellar/llvm/22.1.8/bin/../include/c++/v1/__configuration/availability.h:204:4: error: "The selected platform is no longer supported by libc++." [-Werror,-W#warnings]
  204 | #  warning "The selected platform is no longer supported by libc++."
      |    ^
1 error generated.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@
 cmake_minimum_required(VERSION 3.18.0)
 
 if(APPLE)
-  set(CMAKE_OSX_DEPLOYMENT_TARGET 10.15 CACHE STRING "Minimum OS X deployment version.")
+  set(CMAKE_OSX_DEPLOYMENT_TARGET 14.0 CACHE STRING "Minimum OS X deployment version.")
 endif()
 
 set(PROJECT_NAME libCellML)

